action/install.go

```diff
@@ -25,6 +29,10 @@ func (i *Install) Run(c *claim.Claim, creds credentials.Set, w io.Writer) error
 		return err
 	}
 
+	if i.OperationConfig != nil {
```

Comment:
I originally made the change that way. 😀 It would be a breaking change since it alters the method signature. I wasn't sure if that would be a welcome change(?) so I went with this instead for an initial proposal.

It looked like this:

```
func (i *Install) Run(c *claim.Claim, creds credentials.Set, ops OperationConfig...)
```

Setting the writer on the op was something that the caller managed via the `ops` parameter.

Reply:
Ok I was wondering about that too, but as we didn't hit the 1.0 yet, should we "care" about about breaking changes? I think we should focus on the API design instead 😄

Reply:
Here's what it would look like with @silvin-lubecki 's suggested change. 

```
type OperationConfigFunc func(op *driver.Operation) error

type OperationConfigs []OperationConfigFunc

// ApplyConfig safely applies the configuration function to the operation, if defined,
// and returns any error.
func (cfgs OperationConfigs) ApplyConfig(op *driver.Operation) error {
	var err error
	for _, cfg := range cfgs {
		err = cfg(op)
		if err != nil {
			return err
		}
	}
	return nil
}

// Run performs an installation and updates the Claim accordingly
func (i *Install) Run(c *claim.Claim, creds credentials.Set, w io.Writer, opCfgs ...OperationConfigFunc) error {
	// ...

	err = OperationConfigs(opCfgs).ApplyConfig(op)
	if err != nil {
		return err
	}
```

I'm leaving the writer alone for now just to show what the overall change looks like. It could made into a config func that everyone would have to pass in (more disruptive), or we could keep it as is.

Reply:
@glyn What do you think?